### ISSUE TITLE
Have an exit code `2` for when tests / monitors fail

### ISSUE THREAD

**Issue by JCZuurmond** [sodadata/soda-sql#555]
**Is your feature request related to a problem? Please describe.**
I would like to differentiate between failing tests and other kind of failures (bugs, connections not working, etc.). This would allow me to have a CI running that does not fail if the test / monitors fail (we get notifications for this via the cloud), but the CI **does** fail if the Soda scan fails for another reason.

**Describe the solution you'd like**
Exit with code `2` if tests / monitors fail, exit with code `1` for any other failures.

**Comment by JCZuurmond** [comment]
Optionally, we could even separate tests and monitors. E.g.: fail with exit code `2` if tests fail, fail with exit code `3` if monitors fail and fail with exit code `1` for any other failure.

**Comment by vijaykiran** [comment]
I agree, this makes sense. We don't have a clear distinction between monitor failures and test failures though. That can come later.
